core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -74,23 +84,63 @@ enum class Iterate
 
 template <typename ExecSpace>
 struct default_outer_direction {
-  using type = Iterate;
-#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
-  static constexpr Iterate value = Iterate::Left;
-#else
+  using type                     = Iterate;
   static constexpr Iterate value = Iterate::Right;
+};
+
+#ifdef KOKKOS_ENABLE_CUDA
```

Comment:
should these be in the execution space specific files, so that there is less ifdefing in this one? Just declare the default version above the inclusion of the ExecSpace headers and stick the specializations maybe even into Kokkos_Cuda.hpp etc?

Reply:
Done.